Core/Core/People/ContextUser.swift

```diff
@@ -0,0 +1,78 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import CoreData
+
+public final class ContextUser: NSManagedObject {
+    @NSManaged public var id: String
+    @NSManaged public var name: String
+    @NSManaged public var sortableName: String
+    @NSManaged public var shortName: String
+    @NSManaged public var avatarURL: URL?
+    @NSManaged public var email: String?
+    @NSManaged public var courseID: String?
+    @NSManaged public var groupID: String?
+    @NSManaged public var pronouns: String?
+    @NSManaged public var observerID: String?
+
+    public var enrollments: Set<Enrollment> {
+        Set(managedObjectContext?.all(where: #keyPath(Enrollment.userID), equals: id) ?? [])
+    }
+}
+
+extension ContextUser: WriteableModel {
+    @discardableResult
+    public static func save(_ item: APIUser, in context: NSManagedObjectContext) -> ContextUser {
+        let user: ContextUser = context.first(where: #keyPath(ContextUser.id), equals: item.id.value) ?? context.insert()
+        user.id = item.id.value
+        user.name = item.name
+        user.shortName = item.short_name
+        user.sortableName = item.sortable_name
+        user.email = item.email
+        user.avatarURL = item.avatar_url?.rawValue
+        user.pronouns = item.pronouns
+        if let enrollments = item.enrollments {
+            for item in enrollments {
```

Comment:
It's a little confusing that the function's property is called `item` and the local variable within this loop is also called `item`.

Reply:
renamed.